### PR TITLE
ci(codeql-analysis): upgrade actions/checkout to v3

### PR DESCRIPTION
**Pull request by oscard0m**
## PR Checklist
Upgrade `actions/checkout` to `v3` in `codeql-analysis.yml` 

- [ ] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
 
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
```
Comment (jmcdo29) [inline .github/workflows/codeql-analysis.yml:24]: What are the differences between v2 and v3?

Comment (micalevisk) [reply .github/workflows/codeql-analysis.yml:24]: nothing much https://github.com/actions/checkout/compare/v2...v3

Comment (oscard0m) [reply .github/workflows/codeql-analysis.yml:24]: The breaking change is to use `node_version` 16 by default.

It also adds support for `set-safe-directory` option and upgrades some npm package versions (which could have some security patches).

https://github.com/actions/checkout/releases

**Comment by kamilmysliwiec** [comment]
lgtm

**Pull request merged**